File: LeeJaeWoo/ep1/TicketSeller.java

class TicketSeller implements Seller {
    private boolean busy = false;
    private MoneyOffice moneyOffice;

    TicketSeller(MoneyOffice moneyOffice) {
        this.moneyOffice = moneyOffice;
    }

    @Override
    public boolean isBusy() {
        return busy;
    }

    @Override
    public void sellTo(User user) {
        busy = true;
        moneyOffice.plusMoney(user.buy(moneyOffice.getTicket()));
    }


}

File: ParkKwiNam/chapter1/sellable/TicketSeller.kt
package chapter1.sellable

import chapter1.items.Ticket
import chapter1.people.Person
import chapter1.people.User

class TicketSeller(name: String) : Person(name), TicketSellable {
    override fun sellTo(user: User) {
        println("sell from $this to $user")
        user.buyTicket(getTicket())
    }

    override fun getTicket(): Ticket {
        return Ticket()
    }
}